### DISCUSSION TITLE
Tenant Support

### DISCUSSION THREAD

**Discussion by llamorinlabs** [Z3d0X/filament-logger#115] [Q&A]
```php
The model [Spatie\Activitylog\Models\Activity] does not have a relationship named [tenant].
You can change the relationship being used by passing it to the [ownershipRelationship] argument of the [tenant()] method in configuration. 
You can change the relationship being used per-resource by setting it as the [$tenantOwnershipRelationshipName] static property on the [Z3d0X\FilamentLogger\Resources\ActivityResource] resource class.
```

Should I publish the model for Spatie and the add relationship there or is there any way to modify the logger Resource?

**Comment by akaflipper** [comment]
Hi @llamorinaabs 
just moved from another filament logger .. and went through this. 

What I did was create my own App\Models\Activity which extends \Spatie\Activitylog\Models\Activity implements \Spatie\Activitylog\Contracts\Activity

then if you are using Spatie MultiTenancy:  use UsesTenantConnection in the new Activity Model. 
or apply your own flavour of tenancy / relationships ...

then register the model in ActivityPolicy and AuthServiceProvider and it will use Tenant DB for activity ... 
hth
g
